lib/Convos/Core/User.pm

```diff
@@ -151,19 +152,11 @@ sub validate_password {
   my ($self, $plain) = @_;
 
   return 0 unless $self->password and $plain;
-  return 1 if $self->_bcrypt($plain, $self->password) eq $self->password;
-  return 0;
-}
-
-sub _bcrypt {
-  my ($self, $plain, $settings) = @_;
-
-  unless ($settings) {
-    my $salt = join '', map { chr int rand 256 } 1 .. 16;
-    $settings = BCRYPT_BASE_SETTINGS . Crypt::Eksblowfish::Bcrypt::en_base64($salt);
+  my $valid = $self->_crypt->verify_password($plain, $self->password);
+  if ($valid && $self->_crypt->needs_rehash($plain)) {
+    $self->{password} = $self->_crypt->hash_password($plain);
   }
-
-  Crypt::Eksblowfish::Bcrypt::bcrypt($plain, $settings);
+  return $valid;
```

Comment:
This now returns nothing rather than 0, which I think should be ok (and maybe even a little bit safer).

Reply:
Why is it safer?